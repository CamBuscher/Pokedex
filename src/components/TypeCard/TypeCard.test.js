import React from 'react';
import { TypeCard } from './index'
import {shallow} from 'enzyme'

describe('TypeCard', () => {
  let wrapper
  let mockProps

  beforeEach(() => {
    mockProps = {
      pokemon: [{
        name: 'onyx',
        weight: '75',
        spites: {
          front_default: 'hello.jpg'
        }
      }],
      id: 1,
      name: 'ONYX'
    }
    wrapper = shallow(<TypeCard {...mockProps} />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('fetchPokemon', () => {
    it('should call fetch with correct paramaters', () => {
      
      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        json: () => Promise.resolve({
          name: 'onyx',
          weight: '75',
          sprites: {
            front_default: 'hello.jpg'
          }
        })
      }))

      Promise.resolve(wrapper.simulate('click'))
        .then(() => {
          wrapper.update()
        })
        .then(() => {
          expect(window.fetch).toHaveBeenCalled()
        })
    })

    it('should update card state with new pokemon', () => {

      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        json: () => Promise.resolve({
          name: 'onyx',
          weight: '75',
          sprites: {
            front_default: 'hello.jpg'
          }
        })
      }))

      Promise.resolve(wrapper.simulate('click'))
        .then(() => {
          wrapper.update()
        })
        .then(() => {
          expect(wrapper.state().pokemon).toEqual([{
            name: 'onyx',
            weight: '75',
            sprites: {
              front_default: 'hello.jpg'
            }
          }])
        })
    })

    it('should render a pokemon inside the card after call is made', () => {
      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        json: () => Promise.resolve({
          name: 'onyx',
          weight: '75',
          sprites: {
            front_default: 'hello.jpg'
          }
        })
      }))

      Promise.resolve(wrapper.simulate('click'))
        .then(() => {
          wrapper.update()
        })
        .then(() => {
          const info = wrapper.find('div')

          expect(info.length).toEqual(1)
        })
    })
  })
})