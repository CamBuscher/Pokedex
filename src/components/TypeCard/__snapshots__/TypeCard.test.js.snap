// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TypeCard
    id={1}
    name="ONYX"
    pokemon={
        Array [
            Object {
              "name": "onyx",
              "spites": Object {
                "front_default": "hello.jpg",
              },
              "weight": "75",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          ONYX
</h3>,
        Array [],
      ],
      "className": "typeCard",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ONYX",
        },
        "ref": null,
        "rendered": "ONYX",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            ONYX
</h3>,
          Array [],
        ],
        "className": "typeCard",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ONYX",
          },
          "ref": null,
          "rendered": "ONYX",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
